core/node/app_registry/event_processor.go

```diff
@@ -158,6 +162,11 @@ func (p *MessageToAppProcessor) OnMessageEvent(
 		return
 	}
 
+	// Track message posted event
+	p.analytics.Track(ctx, creator.Hex(), "posted message", map[string]any{
```

Comment:
Instead of passing hex why not pass the address and let the `Track` function handle the formatting? That ensures it is consistent everywhere bacause addresses in hex format can be lower, upper and a mix of lower/upper case if it's checksummed.